src/Symfony/Component/Env/Dotenv.php

```diff
@@ -0,0 +1,380 @@
+            }
+            ++$this->cursor;
+            $this->skipComment();
+        } elseif ('"' === $this->data[$this->cursor]) {
+            ++$this->cursor;
+            while ('"' !== $this->data[$this->cursor] || '\\' === $this->data[$this->cursor - 1]) {
+                $value .= $this->data[$this->cursor];
+                ++$this->cursor;
+
+                if ($this->cursor === $this->end) {
+                    throw new FormatException('Missing quote to end the value', $this->createFormatExceptionContext());
+                }
+            }
+            if ("\n" === $this->data[$this->cursor]) {
+                throw new FormatException('Missing quote to end the value', $this->createFormatExceptionContext());
+            }
+            ++$this->cursor;
+            $this->skipComment();
+            $value = str_replace(array('\\"', '\r', '\n'), array('"', "\r", "\n"), $value);
+        } else {
+            $notQuoted = true;
+            while ($this->cursor < $this->end && "\n" !== $this->data[$this->cursor] && !(' ' === $this->data[$this->cursor - 1] && '#' === $this->data[$this->cursor])) {
+                $value .= $this->data[$this->cursor];
+                ++$this->cursor;
+            }
+            $value = rtrim($value);
+            $this->skipComment();
+        }
+
+        $this->skipEmptyLines();
+
+        $currentValue = $value;
+        if (!$singleQuoted) {
+            $value = $this->resolveVariables($value);
+            $value = $this->resolveCommands($value);
+        }
+
+        if ($notQuoted && $currentValue == $value && preg_match('/\s+/', $value)) {
+            throw new FormatException('A value containing spaces must be surrounded by quotes', $this->createFormatExceptionContext());
+        }
+
+        return $value;
+    }
+
+    private function skipWhitespace()
+    {
+        $this->cursor += strspn($this->data, ' ', $this->cursor);
+    }
+
+    private function skipEmptyLines()
+    {
+        if (preg_match('/(\n+|^#[^\n]*(\n*|$))+/Asm', $this->data, $match, null, $this->cursor)) {
+            $this->moveCursor($match[0]);
+        }
+    }
+
+    private function skipComment()
+    {
+        if (preg_match('/ *#[^\n]*(\n*|$)/Asm', $this->data, $match, null, $this->cursor)) {
+            $this->moveCursor($match[0]);
+        }
+    }
+
+    private function resolveCommands($value)
+    {
+        if (false === strpos($value, '$')) {
+            return $value;
+        }
+
+        $regex = '/
+            (\\\\)?               # escaped with a backslash?
+            \$
+            (?<cmd>
+                \(                # require opening parenthesis
+                ([^()]|\g<cmd>)+  # allow any number of non-parens, or balanced parens (by nesting the <cmd> expression recursively)
+                \)                # require closing paren
+            )
+        /x';
+
+        $env = array_replace($_ENV, $_SERVER, $this->values);
```

Comment:
Is rewriting all the `$_SERVER` values to the  `$_ENV` variable intended here? I can understand why this is done for  `$this->values`.